### DISCUSSION TITLE
Allow Connection only to rinkeby and local 8545

### DISCUSSION THREAD

**Discussion by Hussainzz** [wevm/wagmi#259] [Q&A] [answered]
How can i achieve to allow connection to just rinkeby and my local chains ?

```
(connector: Connector) => Promise<{
  data?: {
    account?: string
    chain?: { id: number; unsupported: boolean }
    provider?: Provider
  }
  error?: Error
}>
```
can someone explain me what is that **unsupported** value used for ?

**Comment by tmm** [comment, marked as answer]
You can’t completely restrict the chain a wallet is connected to (since wallets can choose whatever chain they want), but you can set [supported chains for connectors](https://wagmi.sh/docs/connectors/injected#chains-optional). When someone connects or uses a chain that is not specified, `unsupported` is set to `true`. You can use this value to restrict access or notify the user to switch chains (`useNetwork` can help with this).

**Reply by Hussainzz** [reply]
Sorry for the dumb question, can you brief me on difference between a connector and a provider.
do to restrict certain networks do i also need to pass the connectors prop?
why is getDefaultProvider used
```
const provider = providers.getDefaultProvider("http://127.0.0.1:8545",{
  infura: 'someKey',
  alchemy: 'someKey'
});

 <WagmiProvider provider={provider}>
 <WagmiProvider />
```
